tools/clang/tools/dxcompiler/dxcdisassembler.cpp

```diff
@@ -1294,6 +1294,7 @@ void PrintPipelineStateValidationRuntimeInfo(const char *pBuffer,
 
   const unsigned offset = sizeof(unsigned);
   const PSVRuntimeInfo0 *pInfo = (const PSVRuntimeInfo0 *)(pBuffer + offset);
+  const PSVRuntimeInfo2 *pInfo2 = (const PSVRuntimeInfo2 *)(pBuffer + offset);
```

Comment:
The first DWORD that is skipped here by `offset = sizeof(unsigned)` is the size of the `PSVRuntimeInfo` structure.  Only if it is `>= sizeof(PSVRuntimeInfo2)` should you be casting it to that.

In fact, this should use `DxilPipelineStateValidation`, since it handles all this.  Here's how:
```
DxilPipelineStateValidation PSV;
PSV.InitFromPSV0(pBuffer, uBufferSize);
const PSVRuntimeInfo0 *pInfo0 = PSV.GetPSVRuntimeInfo0();
const PSVRuntimeInfo2 *pInfo2 = PSV.GetPSVRuntimeInfo2();
...
if (pInfo2) {
  OS << comment << " Numthreads: (" << pInfo2->NumThreadsX << "," << pInfo2->NumThreadsY << "," << pInfo2->NumThreadsZ << ")\n";
}
```

This requires that size be passed in here, which should be available from the part size when `PSV0` part was looked up.